onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -149,6 +151,14 @@ CastFromString(const std::string& input, DstType& output) {
   output = static_cast<DstType>(intermediate);
 }
 
+template <typename DstType>
+typename std::enable_if<IsOrtFloat8Type<DstType>::value, void>::type
+CastFromString(const std::string& input, DstType& output) {
```

Comment:
nit: this could be consolidated with the IsOrtFloat16Type implementation to reduce source code

Reply:
done